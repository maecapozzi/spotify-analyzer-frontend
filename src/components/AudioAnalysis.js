import React from 'react'
import injectSheet from 'react-jss'
import PropTypes from 'prop-types'

const styles = {
  value: {
    color: '#00C457'
  }
}

const AudioAnalysis = ({classes, type, value}) => (
  <div>
    <p>
      <span>{type}: </span>
      <span className={classes.value}>{value}</span>
    </p>
  </div>
)

AudioAnalysis.propTypes = { 
  type: PropTypes.string,
  value: PropTypes.number
}

export default injectSheet(styles)(AudioAnalysis)
