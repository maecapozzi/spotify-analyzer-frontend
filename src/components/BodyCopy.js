import React from 'react'
import injectSheet from 'react-jss'
import PropTypes from 'prop-types'

const styles = {
  h3: {
    color: '#5A6575',
    fontFamily: 'Inconsolata',
    fontWeight: '300',
    letterSpacing: '2px',
    padding: [20, 20],
    textAlign: 'center'
  }
}

const BodyCopy = ({ classes, string }) => (
  <h3 className={classes.h3}>{string}</h3>
)

BodyCopy.propTypes = {
  string: PropTypes.string
}

export default injectSheet(styles)(BodyCopy)
