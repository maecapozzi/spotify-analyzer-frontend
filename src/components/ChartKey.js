import React from 'react'

const ChartKey = () => (
  <div>
    <p>Danceability: "A measure from 0.0 to 1.0 of "how suitable a track is for dancing."</p>
    <p>Acousticness: "A confidence measure from 0.0 to 1.0 of whether the track is acoustic."</p>
    <p>Energy: "A measure from 0.0 to 1.0 that represents a perceptual measure of intensity and activity."</p>
    <p>Liveness: "Detects the presence of an audience in the recording."</p>
    <p>Instrumentalness: "Predicts whether a track contains no vocals."</p>
    <p>Speechiness: "Speechiness detects the presence of spoken words in a track."</p>
    <p>Valence: "A measure from 0.0 to 1.0 describing the musical positiveness conveyed by a track."</p>
  </div>
)

export default ChartKey
