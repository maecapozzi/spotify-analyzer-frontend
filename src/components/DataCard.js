import React from 'react'
import injectSheet from 'react-jss'
import PropTypes from 'prop-types'

import Card from 'material-ui/Card'

const styles = {
  card: {
    border: '1px #272B2D solid',
    padding: '2em',
    backgroundColor: '#181A1B !important',
    height: '430px'
  }
}

const DataCard = ({ classes, component }) => {
  return <Card className={classes.card}>{component}</Card>
}

DataCard.propTypes = {
  component: PropTypes.object
}
export default injectSheet(styles)(DataCard)
