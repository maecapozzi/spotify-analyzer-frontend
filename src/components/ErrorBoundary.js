import React, { Component } from 'react'
import injectSheet from 'react-jss'
import PropTypes from 'prop-types'

const styles = {
  h1: {
    color: '#BFDBF7',
    fontFamily: 'Inconsolata',
    textAlign: 'center'
  }
}

class ErrorBoundary extends Component {
  constructor (props) {
    super(props)
    this.state = { hasError: false }
  }

  componentDidCatch (error, info) {
    if (error) {
      this.setState(state => ({ ...state, hasError: true }))
    }
  }

  render () {
    if (this.state.hasError) {
      return <h1 className={this.props.classes.h1}>Something went wrong.</h1>
    }
    return this.props.children
  }
}

ErrorBoundary.propTypes = {
  classes: PropTypes.object,
  children: PropTypes.object
}

export default injectSheet(styles)(ErrorBoundary)
