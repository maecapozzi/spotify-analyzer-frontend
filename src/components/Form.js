import React, { Component } from 'react'
import axios from 'axios'

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {value: ''};

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({value: event.target.value});
  }

  handleSubmit(event) {
    axios.get('/search?', {
      params: {
        track: this.state.value
      }
    })
    .then(function (response) {
      console.log(response);
    })
    .catch(function (error) {
      console.log(error);
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input type="text" value={this.state.value} onChange={this.handleChange} />
        <input type="submit" value="Submit" />
      </form>
    );
  }
}

export default Form
