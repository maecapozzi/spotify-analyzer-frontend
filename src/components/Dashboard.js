import React, { Component } from 'react'
import ReactHighcharts from 'react-highcharts'
import TrackData from './TrackData'
import ChartKey from './ChartKey'
import { Grid, Row, Col } from 'react-flexbox-grid'
import AudioAnalysis from './AudioAnalysis'
import DataCard from './DataCard'

class Dashboard extends Component {
  render () {
    const config = {
      chart: {
        type: 'bar',
        backgroundColor: '#181A1B'
      },
      labels: {
        color: '#5A6575'
      },
      title: 'Audio Features',
      xAxis: {
        categories: ['danceability', 'acousticness', 'energy', 'liveness', 'instrumentalness', 'speechiness', 'valence']
      },
      yAxis: {
        max: 1,
        title: {
          text: null
        }
      },
      plotOptions: {
        series: {
          colorByPoint: true,
          colors: ['#00C457']
        }
      },
      series: [{
        data: [
          this.props.danceability,
          this.props.acousticness,
          this.props.energy,
          this.props.liveness,
          this.props.instrumentalness,
          this.props.speechiness,
          this.props.valence
        ]
      }]
    }
    return (
      <div>
        <Grid fluid>
          <Row>
            <Col xs={12} lg={12}>
              <DataCard component={<ReactHighcharts config={config} ref='test' />} />
            </Col>
          </Row>
          <Row>
            <Col xs={12} lg={6}>
              <DataCard component={
                <TrackData
                  title={this.props.title}
                  artist={this.props.artist}
                  albumLink={this.props.albumLink}
                  popularity={this.props.popularity}
                  albumImages={this.props.albumImages}
                />
              } />
            </Col>
            <Col xs={12} lg={6}>
              <DataCard component={<ChartKey />} />
            </Col>
          </Row>
          <Row>
            <Col xs={12} lg={4}>
              <DataCard
                component={
                  <AudioAnalysis
                    type={'Time Signature'} value={this.props.timeSignature}
                  />}
              />
            </Col>
            <Col xs={12} lg={4}>
              <DataCard component={
                <AudioAnalysis
                  type={'Tempo'} value={this.props.tempo}
                />
              } />
            </Col>
            <Col xs={12} lg={4}>
              <DataCard component={
                <AudioAnalysis
                  type={'Duruation'} value={this.props.duration}
                />
              } />
            </Col>
          </Row>
        </Grid>
      </div>
    )
  }
}

export default Dashboard
