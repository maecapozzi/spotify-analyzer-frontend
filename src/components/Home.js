import React, { Component } from 'react'

import { connect } from 'react-redux'
import axios from 'axios'
import PropTypes from 'prop-types'
import BodyCopy from './BodyCopy'
import Dashboard from './Dashboard'
import Header from './Header'
import SearchBar from 'material-ui-search-bar'
import SearchResults from './SearchResults'
// Todo: Refactor getSong & getSongs into redux
import { getSong } from '../lib/apiService'
import { getSongs } from '../lib/searchHelper'
import { collectUserInput } from '../actions/index'

export class Home extends Component {
  constructor (props) {
    super(props)
    this.url = `${this.props.url}/search?`
    this.state = {
      loadingPage: false,
      results: '',
      danceability: '',
      acousticness: '',
      energy: '',
      liveness: '',
      instrumentalness: '',
      loudness: '',
      speechiness: '',
      valence: '',
      title: '',
      artist: '',
      showSearchResults: true
    }

    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.setLoadingPage = this.setLoadingPage.bind(this)
    this.setHome = this.setHome.bind(this)
    this.setDashboard = this.setDashboard.bind(this)
    this.setLoadingScreen = this.setLoadingScreen.bind(this)
  }

  handleChange (event) {
    this.props.dispatch(collectUserInput(event))
  }

  handleSubmit () {
    const value = this.props.value
    getSong(value)
      .then(response => {
        if (response.status === 200) {
          const results = getSongs(response.data.results)
          this.setState({ results, showSearchResults: true })
        }
      })
      .catch(error => {
        if (error) {
          throw new Error('Could not get a response from the server!')
        }
      })
  }

  setLoadingPage () {
    this.setState({ loadingPage: true })
    setTimeout(() => this.setState({ loadingPage: false }), 6000)
  }

  handleClick (result) {
    this.setLoadingPage()
    let uri = this.props.url + '/analyze/' + result.id
    axios
      .get(uri, {
        withCredentials: true
      })
      .then(response => {
        if (response.status === 200) {
          const responseData = {
            audioFeatures: response.data[0],
            trackData: response.data[1],
            audioAnalysis: response.data[2]
          }
          this.setState({
            title: result.name,
            artist: result.artist,
            danceability: responseData.audioFeatures.danceability,
            acousticness: responseData.audioFeatures.acousticness,
            energy: responseData.audioFeatures.energy,
            liveness: responseData.audioFeatures.liveness,
            instrumentalness: responseData.audioFeatures.instrumentalness,
            loudness: responseData.audioFeatures.loudness,
            speechiness: responseData.audioFeatures.speechiness,
            valence: responseData.audioFeatures.valence,
            showSearchResults: false,
            albumLink: responseData.trackData.album.external_urls.spotify,
            popularity: responseData.trackData.popularity,
            albumImages: responseData.trackData.album.images,
            timeSignature: responseData.audioAnalysis.track.time_signature,
            tempo: responseData.audioAnalysis.track.tempo,
            songKey: responseData.audioAnalysis.track.key,
            duration: responseData.audioAnalysis.track.duration
          })
        }
      })
      .catch(error => {
        console.log(error)
      })
  }

  setHome () {
    return (
      <div>
        <Header string={'EARWORM'} />
        <BodyCopy
          string={
            'Search for a song on Spotify and view the audio features of that song'
          }
        />
        <SearchBar
          style={{
            margin: '0 auto',
            maxWidth: 800,
            marginTop: 50
          }}
          onSubmit={this.handleSubmit}
          onChange={this.handleChange}
          onRequestSearch={this.handleSubmit}
        />
      </div>
    )
  }

  setDashboard () {
    return (
      <div>
        {this.setHome()}
        <div className='song-data'>
          <Dashboard
            title={this.state.title}
            artist={this.state.artist}
            danceability={this.state.danceability}
            acousticness={this.state.acousticness}
            energy={this.state.energy}
            liveness={this.state.liveness}
            instrumentalness={this.state.instrumentalness}
            speechiness={this.state.speechiness}
            valence={this.state.valence}
            albumLink={this.state.albumLink}
            popularity={this.state.popularity}
            albumImages={this.state.albumImages}
            timeSignature={this.state.timeSignature}
            tempo={this.state.tempo}
            songKey={this.state.songKey}
            loudness={this.state.loudness}
            duration={this.state.duration}
          />
        </div>
      </div>
    )
  }

  setLoadingScreen () {
    return <h1 className='loading-data'>Loading Data</h1>
  }

  render () {
    if (!this.state.results) {
      return this.setHome()
    } else if (this.state.loadingPage) {
      return this.setLoadingScreen()
    } else if (this.state.showSearchResults) {
      return (
        <div>
          {this.setHome()}
          <SearchResults
            handleClick={this.handleClick.bind(this)}
            label='Submit'
            results={this.state.results}
            url={this.props.url}
          />
        </div>
      )
    } else {
      return this.setDashboard()
    }
  }
}

Home.propTypes = {
  url: PropTypes.string.isRequired
}

const mapStateToProps = state => {
  return {
    value: state.value
  }
}

export default connect(mapStateToProps)(Home)
