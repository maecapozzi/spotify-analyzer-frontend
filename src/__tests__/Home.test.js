import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { Home } from '../components/Home'

configure({ adapter: new Adapter() })

test('it renders correctly', () => {
  const component = shallow(<Home />)
  expect(component).toMatchSnapshot()
})
