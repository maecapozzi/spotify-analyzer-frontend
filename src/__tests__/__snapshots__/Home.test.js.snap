// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Jss(Header)
          string="earworm"
/>,
        <h3
          className="subheader"
>
          Search for a song on Spotify and view the audio features of that song.
</h3>,
        <SearchBar
          closeIcon={
                    <NavigationClose
                              color="#9e9e9e"
                    />
          }
          dataSource={Array []}
          dataSourceConfig={
                    Object {
                              "text": "text",
                              "value": "value",
                            }
          }
          disabled={false}
          hintText="Search"
          onChange={[Function]}
          onRequestSearch={[Function]}
          onSubmit={[Function]}
          searchIcon={
                    <ActionSearch
                              color="#9e9e9e"
                    />
          }
          spellCheck={false}
          style={
                    Object {
                              "margin": "0 auto",
                              "marginTop": 50,
                              "maxWidth": 800,
                            }
          }
          value=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "string": "earworm",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for a song on Spotify and view the audio features of that song.",
          "className": "subheader",
        },
        "ref": null,
        "rendered": "Search for a song on Spotify and view the audio features of that song.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeIcon": <NavigationClose
            color="#9e9e9e"
/>,
          "dataSource": Array [],
          "dataSourceConfig": Object {
            "text": "text",
            "value": "value",
          },
          "disabled": false,
          "hintText": "Search",
          "onChange": [Function],
          "onRequestSearch": [Function],
          "onSubmit": [Function],
          "searchIcon": <ActionSearch
            color="#9e9e9e"
/>,
          "spellCheck": false,
          "style": Object {
            "margin": "0 auto",
            "marginTop": 50,
            "maxWidth": 800,
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Jss(Header)
            string="earworm"
/>,
          <h3
            className="subheader"
>
            Search for a song on Spotify and view the audio features of that song.
</h3>,
          <SearchBar
            closeIcon={
                        <NavigationClose
                                    color="#9e9e9e"
                        />
            }
            dataSource={Array []}
            dataSourceConfig={
                        Object {
                                    "text": "text",
                                    "value": "value",
                                  }
            }
            disabled={false}
            hintText="Search"
            onChange={[Function]}
            onRequestSearch={[Function]}
            onSubmit={[Function]}
            searchIcon={
                        <ActionSearch
                                    color="#9e9e9e"
                        />
            }
            spellCheck={false}
            style={
                        Object {
                                    "margin": "0 auto",
                                    "marginTop": 50,
                                    "maxWidth": 800,
                                  }
            }
            value=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "string": "earworm",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for a song on Spotify and view the audio features of that song.",
            "className": "subheader",
          },
          "ref": null,
          "rendered": "Search for a song on Spotify and view the audio features of that song.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeIcon": <NavigationClose
              color="#9e9e9e"
/>,
            "dataSource": Array [],
            "dataSourceConfig": Object {
              "text": "text",
              "value": "value",
            },
            "disabled": false,
            "hintText": "Search",
            "onChange": [Function],
            "onRequestSearch": [Function],
            "onSubmit": [Function],
            "searchIcon": <ActionSearch
              color="#9e9e9e"
/>,
            "spellCheck": false,
            "style": Object {
              "margin": "0 auto",
              "marginTop": 50,
              "maxWidth": 800,
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
